src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -89,10 +92,16 @@
 public class BigQueryPdao implements PrimaryDataAccess {
     private static final Logger logger = LoggerFactory.getLogger(BigQueryPdao.class);
 
+    private static final int DEFAULT_MAX_NUM_UPDATE_LIMIT_INDUCED_RETRIES = 3;
+    private static final int DEFAULT_RETRY_WAIT_MS = 500;
+
     private final String datarepoDnsName;
     private final DataLocationService dataLocationService;
     private final DatasetService datasetService;
 
+    @Value("${datarepo.bq.rateLimitRetries}") private Integer rateLimitRetries;
+    @Value("${datarepo.bq.rateLimitRetryWaitMs}") private Integer rateLimitWait;
+
```

Comment:
why not use the config enum for these guys?

Reply:
Ooh...what's the config enum? It does sound like I thing I should use :)

Reply:
We never use $Value. It is considered harmful: harder to find the source of configuration, not well typed, ...
Instead, we use configuration classes that map to specific properties. Then use getters to access those classes.
See `main/java/bio/terra/app/configuration/ApplicationConfiguration` for the main one.

If you want to have run-time configurability of a config item, then you need to do additional work that Rori referred to:
* add an entry to `.../service/configuration/ConfiguEnum.java`
* add a mapping in `.../service/configuration/ConfigurationService.java` toward the bottom of the file.

You will see in some of our tests that we manipulate parameters by hitting a configuration endpoint - the same one used to enable faults.

Reply:
Got it. I'll add a new config object for the BigQuery config.  I'm going to hold off for the runtime config...that seems to be geared towards test parameters.  Also not sure that it's a runtime change.  Seems reasonable to have these effectively tied to the deployment